File: Nootbook/app/src/main/java/com/example/nootbook/Signup_activity.java
package com.example.nootbook;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ProgressBar;
import android.widget.TextView;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.Firebase;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

public class Signup_activity extends AppCompatActivity {

    FirebaseAuth auth;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_signup);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
        auth = FirebaseAuth.getInstance();

        EditText fullname_input = findViewById(R.id.register_fullname);
        EditText email_input = findViewById(R.id.register_email);
        EditText password_input = findViewById(R.id.register_password);
        Button singupBtn = findViewById(R.id.signup_button);


        singupBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String fullname = fullname_input.getText().toString();
                String email = email_input.getText().toString();
                String password = password_input.getText().toString();
                    if (fullname.isEmpty()){
                        fullname_input.setError("Full Name Is Required");
                        return;
                    }
                    if (email.isEmpty()){
                        email_input.setError("Email Is Required");
                        return;
                    }
                    if (password.isEmpty()){
                        password_input.setError("Password Is Required");
                        return;
                    }
                singupBtn.setEnabled(false);
                   singupBtn.setText("Processing..");
                ProgressBar loader = findViewById(R.id.loader);
                loader.setVisibility(View.VISIBLE);
                auth.createUserWithEmailAndPassword(email,password).addOnCompleteListener(Signup_activity.this, new OnCompleteListener<AuthResult>() {
                    @Override
                    public void onComplete(@NonNull Task<AuthResult> task) {
                        singupBtn.setEnabled(true);
                        singupBtn.setText(R.string.ragisternow);
                        loader.setVisibility(View.GONE);
                      Exception  exce =task.getException();
                       if (exce.getMessage()!=null){
                           String msg = exce.getMessage();
                           Toast.makeText(Signup_activity.this, msg, Toast.LENGTH_LONG).show();
                           return;
                       }

                        Intent switchtoNoteintent =  new Intent(Signup_activity.this,Noteactivty.class);
                       startActivity(switchtoNoteintent);
                       finish();
                    }
                });
            }
        });




//        switch to  loginactivity
        TextView switch_btn = findViewById(R.id.switch_to_sign_in);
        switch_btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(Signup_activity.this, Loginactivity.class);
                startActivity(intent);
                finish();
            }
        });

        FirebaseUser session = auth.getCurrentUser();
        if (session!=null){
            Intent intent = new Intent(Signup_activity.this,Noteactivty.class);
            startActivity(intent);
            finish();
        }

    }
}

File: MistriBaba/app/src/main/java/com/example/mistribaba/SignupActivity.kt
package com.example.mistribaba

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.widget.TextView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.android.volley.Request
import com.android.volley.toolbox.JsonObjectRequest
import com.android.volley.toolbox.Volley
import org.json.JSONObject
import kotlin.apply

class SignupActivity : AppCompatActivity() {

    private var alertDialog: AlertDialog? = null

    private fun showAlertDialog(context: Context, title: String, message: String) {
        val builder = AlertDialog.Builder(context)
        builder.setTitle(title)
        builder.setMessage(message)
        builder.setPositiveButton("OK") { dialog, _ ->
            dialog.dismiss()
        }
        builder.show()
    }
    private fun showLoading(context: Context, title: String, message: String) {
        val builder = AlertDialog.Builder(context)
        builder.setTitle(title)
        builder.setMessage(message)
        builder.setCancelable(false)
        alertDialog = builder.create()
        alertDialog?.show()
    }

    private fun dismissLoading() {
        alertDialog?.dismiss()
    }
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_signup)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        val reDirect_SignInPage = findViewById<TextView>(R.id.reDirect_SignInPage);
        reDirect_SignInPage.setOnClickListener {
            val intent = Intent(this, MainActivity::class.java)
            startActivity(intent)
        }



        var registerBtn = findViewById<TextView>(R.id.registerBtn);
        registerBtn.setOnClickListener {
            var phone = findViewById<TextView>(R.id.phone);
            var Name = findViewById<TextView>(R.id.Name);
            var password = findViewById<TextView>(R.id.password);
            var cPassword = findViewById<TextView>(R.id.cPassword);
            if(phone.text.isEmpty() || Name.text.isEmpty() || password.text.isEmpty() || cPassword.text.isEmpty()){
                showAlertDialog(this, "Registration Failed", "Please Enter All Fields")
                return@setOnClickListener
            }
            if(password.text.toString() != cPassword.text.toString()){
                showAlertDialog(this, "Password Mismatch", "Password Mismatch")
                return@setOnClickListener
            }
            var data = JSONObject().apply {
                put("mobilenumber", phone.text.toString())
                put("fullname", Name.text.toString())
                put("userpassword", password.text.toString())
                put("usertype","2")
            };
            var req = Volley.newRequestQueue(this);
            showLoading(this, "Please Wait", "Loading...")
            var jar = JsonObjectRequest(Request.Method.POST,
                userinfo.backendurl+"/api/register",
                data,
                {
                    response->
                    dismissLoading();
                    showAlertDialog(this, "Registration Successful", "Registration Successful")
                },
                {
                    error->;
                    dismissLoading();
                    showAlertDialog(this, "Registration Failed", "Registration failed: ${error.message}")

                }

            );
            req.add(jar);


        }

    }
}